docs/reactivity.md

```diff
@@ -0,0 +1,410 @@
+
+A [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) represents the result of an asynchronous operation: some data loaded from a file, say, or a module imported dynamically, or the end of an animation, or some text input from the user.
+
+In Framework, when one code block refers to a promise defined in another code block, the referencing code block implicitly awaits the promise. This means promises are often invisible — you don’t have to worry whether something is a promise or whether it’s resolved.
+
+<div class="note">Implicit <code>await</code> only applies <i>across</i> code blocks, not <i>within</i> a code block. Within a code block, a promise is just a promise.</div>
+
+For example, below `FileAttachment.json` returns a promise, and so the value of `volcano` is a promise.
+
+```js echo
+const volcano = FileAttachment("volcano.json").json();
+```
+
+And yet if we reference `volcano` in another code block or inline expression, we don’t need to `await`. The `await` is implicit; the code block automatically waits for the `volcano` promise to resolve before running.
+
+```js echo
+volcano
+```
+
+The volcano dataset has ${volcano.values.length.toLocaleString("en-US")} values.
+
+```md run=false
+The volcano dataset has ${volcano.values.length.toLocaleString("en-US")} values.
+```
+
+This pattern is especially useful for loading multiple files. The files are loaded in parallel, and referencing code blocks only wait for the files they need. This is faster than loading files sequentially with `await`, and simpler than `Promise.all`.
+
+```js run=false
+const a = FileAttachment("a.csv").csv({typed: true});
+const b = FileAttachment("b.csv").csv({typed: true});
+const c = FileAttachment("c.csv").csv({typed: true});
+```
+
+Implicit await causes the entire code block to wait, not just expressions that reference promises. This distinction is usually invisible, but you might notice if you have a code block that references both “slow” and “fast” promises: the code block waits for all promises to resolve, and thus is gated by the slowest promise. Below, “fast” and “slow” are printed at the same time.
+
+```js echo
+const fast = new Promise((resolve) => setTimeout(() => resolve("fast"), 500));
+const slow = new Promise((resolve) => setTimeout(() => resolve("slow"), 5000));
+```
+```js echo
+display(fast);
+display(slow);
+```
+
+Implicit await means waiting for the defining code block to fulfill, not only the referenced promises. Below, even though `one` and `two` are defined synchronously, the referencing code block must wait 5 seconds for the defining block to resolve. This allows a defining code block to initialize state asynchronously while preventing referencing code blocks from seeing partially-initialized state.
+
+```js echo
+const one = 1;
+const two = 2;
+await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
There’s an explicit await in this code block, but there’s no explicit await when referencing `one` and `two` in the code block below. And that other code block still has to wait for this promise to resolve before it can see the values of `one` and `two`.